ui/app/models/cluster.js

```diff
@@ -5,75 +5,104 @@
     }
-  }),
+  }
 
-  sealThreshold: alias('leaderNode.sealThreshold'),
-  sealProgress: alias('leaderNode.progress'),
-  sealType: alias('leaderNode.type'),
-  storageType: alias('leaderNode.storageType'),
-  hcpLinkStatus: alias('leaderNode.hcpLinkStatus'),
-  hasProgress: gte('sealProgress', 1),
-  usingRaft: equal('storageType', 'raft'),
+  get sealThreshold() {
+    return this.leaderNode?.sealThreshold;
+  }
+  get sealProgress() {
+    return this.leaderNode?.progress;
+  }
+  get sealType() {
+    return this.leaderNode?.type;
+  }
+  get storageType() {
+    return this.leaderNode?.storageType;
+  }
+  get hcpLinkStatus() {
+    return this.leaderNode?.hcpLinkStatus;
+  }
+  get hasProgress() {
+    return this.sealProgress >= 1;
+  }
+  get usingRaft() {
+    return this.storageType === 'raft';
+  }
 
   //replication mode - will only ever be 'unsupported'
   //otherwise the particular mode will have the relevant mode attr through replication-attributes
-  mode: attr('string'),
-  allReplicationDisabled: and('{dr,performance}.replicationDisabled'),
-  anyReplicationEnabled: or('{dr,performance}.replicationEnabled'),
+  @attr('string') mode;
+  get allReplicationDisabled() {
+    return this.dr?.replicationDisabled && this.performance?.replicationDisabled;
+  }
+  get anyReplicationEnabled() {
+    return this.dr?.replicationEnabled || this.performance?.replicationEnabled;
+  }
 
-  dr: belongsTo('replication-attributes', { async: false, inverse: null }),
-  performance: belongsTo('replication-attributes', { async: false, inverse: null }),
+  @belongsTo('replication-attributes', { async: false, inverse: null }) dr;
+  @belongsTo('replication-attributes', { async: false, inverse: null }) performance;
   // this service exposes what mode the UI is currently viewing
   // replicationAttrs will then return the relevant `replication-attributes` model
-  rm: service('replication-mode'),
-  drMode: alias('dr.mode'),
-  replicationMode: alias('rm.mode'),
-  replicationModeForDisplay: computed('replicationMode', function () {
+  @service('replication-mode') rm;
+  get drMode() {
+    return this.dr.mode;
+  }
+  get replicationMode() {
+    return this.rm.mode;
+  }
+  get replicationModeForDisplay() {
     return this.replicationMode === 'dr' ? 'Disaster Recovery' : 'Performance';
-  }),
-  replicationIsInitializing: computed('dr.mode', 'performance.mode', function () {
+  }
+  get replicationIsInitializing() {
     // a mode of null only happens when a cluster is being initialized
     // otherwise the mode will be 'disabled', 'primary', 'secondary'
-    return !this.dr.mode || !this.performance.mode;
-  }),
-  replicationAttrs: computed('dr.mode', 'performance.mode', 'replicationMode', function () {
+    return !this.dr?.mode || !this.performance?.mode;
+  }
+  get replicationAttrs() {
     const replicationMode = this.replicationMode;
     return replicationMode ? get(this, replicationMode) : null;
```

Comment:
I tried that, and since `replicationMode` is a dynamic string `get` is the only way i got this to work. It would have to be something like `this[replicationMode]` which, in testing, didn't work. I also learned that `get` is the only way to get nested paths, so if something was at `foo.bar.baz` you could do `get(foo, "bar.baz")` but not `foo["bar.baz"]`

Reply:
Ah okay - thanks for explaining!